.github/workflows/discover-lean-pr-testing.yml

```
@@ -0,0 +1,99 @@
+name: Discover `lean-pr-testing` branches
+
+on:
+  push:
+    branches:
+      - nightly-testing
+    paths:
+      - lean-toolchain
+
+jobs:
+  discover-lean-pr-testing:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout mathlib repository
+      uses: actions/checkout@v3
+      with:
+        repository: leanprover-community/mathlib
+        ref: nightly-testing
+        fetch-depth: 0  # Fetch all branches
+
+    - name: Set up Git
+      run: |
+        git config --global user.name "github-actions"
+        git config --global user.email "github-actions@github.com"
+
+    - name: Read date from lean-toolchain
+      id: fetch-date
+      run: |
+        TODAY=$(grep -oP 'nightly-\K\d{4}-\d{2}-\d{2}' lean-toolchain)
+        echo "TODAY=$TODAY" >> "$GITHUB_ENV"
+
+    - name: Set YESTERDAY
+      id: set-yesterday
+      run: |
+        YESTERDAY=$(date -d "$TODAY -1 day" +%Y-%m-%d)
+        echo "YESTERDAY=$YESTERDAY" >> "$GITHUB_ENV"
+
+    - name: Clone lean4-nightly repository and get PRs
+      id: get-prs
+      run: |
+        NIGHTLY_URL="https://github.com/leanprover/lean4-nightly.git"
+        
+        # Create a temporary directory for cloning
+        cd "$(mktemp -d)"
```

```suggestion
        cd "$(mktemp -d)" || exit 1
```
These guards are possibly superfluous, but act as stopgags in case something goes wrong and help getting better error messages.

Same comment on the `cd lean4-nightly` below that is probably more important!